### PR TITLE
Query cli update

### PR DESCRIPTION
**Pull request by Neketek**
So, I changed everything substantially. Because I needed to organize everything to work better with the tree-like structure of the query command. Therefore now commands can be invoked in the next way:

 ```cot query <blueprint_generation_options...> <<command|group> <command_options...> <command_args...>...>```

Basically it's a ```command->subcommand``` approach. 

Further changes on the same scale are highly improbable because the current approach is very flexible.

**Commands tree structure:**
1. list - group:
    1. tiers - command
    1. components - command
    1. occurrences - command
1. describe - group:
    1. occurrence - command/group(can be invoked):
          1. attributes - command
          1. settings - command
          1. resources - command
          1. solution - command
          1. get - command
1. get - command

@roleyfoley Please, check the correctness of the output on the production blueprints.

### CODE REVIEW

**Comment by roleyfoley** [comment]
Tested with a live CMDB and all seems to be working as expected. Happy with the flow of the commands as well

File: gen3-cli/cot/command/query/__init__.py [hunk 1]
```diff
@@ -125,12 +44,327 @@ def components_table(data):
...
+    )
+    click.echo(json.dumps(result, indent=4))
+
+
+@query_group.group('describe')
+@click.pass_context
+def describe_group(ctx):
+    """
+    Queries category
+    """
```
Comment (koriaf) [inline gen3-cli/cot/command/query/__init__.py]: this looks strange

File: gen3-cli/cot/command/query/__init__.py [hunk 2]
```diff
@@ -125,12 +44,327 @@ def components_table(data):
...
+    )
+    click.echo(json.dumps(result, indent=4))
+
+
+@query_group.group('list')
+@click.pass_context
+def list_group(ctx):
+    """
+    Queries category
+    """
```
Comment (koriaf) [inline gen3-cli/cot/command/query/__init__.py]: and this; strange it's not a syntax error at all

Comment (Neketek) [reply gen3-cli/cot/command/query/__init__.py]: And why it should be a syntax error? Just curious because it's my preferred way to align docstrings.

Comment (koriaf) [reply gen3-cli/cot/command/query/__init__.py]: It just looks like some function without a body

**Pull request merged**